File: 4. Sorting Algo's/Selection Sort/SelectionSort.java
import java.util.*;

public class SelectionSort {
    public static void selectionSort(int[] arr){
        int l = arr.length;
        for(int i = 0; i < l-1; i++){
            int minidx = i;
            for(int cur = i + 1; cur < l; cur++ ){
                if(arr[cur] < arr[minidx]){
                    minidx = cur;
                }
            }
            int temp = arr[i];
            arr[i] = arr[minidx];
            arr[minidx] = temp;
        }
    }
    public static void main(String[] args){
        int[] arr = {9, 8, 3, 1, 4, 6, 2};
        selectionSort(arr);
        for(int num : arr){
            System.out.println(num + " ");
        }
    }
}
/*  
Another way to write

import java.util.Arrays;

public class SelectionSortExample {

    public int[] selectionSort(int[] arr) {
        int n = arr.length;

        for (int i = 0; i < n - 1; i++) {
            int minIndex = i;

            for (int j = i + 1; j < n; j++) {
                if (arr[j] < arr[minIndex]) {
                    minIndex = j;
                }
            }

            int temp = arr[i];
            arr[i] = arr[minIndex];
            arr[minIndex] = temp;
        }

        return arr;
    }
    public static void main(String[] args) {
        int[] arr = {64, 25, 12, 22, 11};

        --int[] sortedArr = selectionSort(arr);-- only this line if there was static, we could directly call it in our main
        
        SelectionSorter sorter = new SelectionSorter(); --these two lines required if theres NO static, gotta make a object--
        int[] sortedArr = sorter.selectionSort(arr);

        System.out.println(Arrays.toString(sortedArr));
    }
}
Important thing to note, here arr and sortedArr prints the same sorted array
*/

File: 4. Sorting Algo's/Selection Sort/SelectionSort.py
def maxselectionsort(arr):






    pass
def minselectionsort(arr):
    l = len(arr) 
    for i in range(l - 1):
        minidx = i
        for curr in range(i + 1, l):
            if arr[curr] < arr[minidx]:
                minidx = curr
        arr[i], arr[minidx] = arr[minidx], arr[i]
        """ 
        we write it this way in java Because Java does not support Python-style multiple assignment.
        temp = arr[i]
        arr[i] = arr[minidx]
        arr[minidx] = temp
        """


arr = [9,5,2,8,1]
minselectionsort(arr)
print(arr)  
"""
class SelectionSort:
    def selection_sort(self, arr):
        n = len(arr)

        for i in range(n - 1):
            min_index = i

            for j in range(i + 1, n):
                if arr[j] < arr[min_index]:
                    min_index = j

            arr[i], arr[min_index] = arr[min_index], arr[i]

        return arr


if __name__ == "__main__":
    arr = [64, 25, 12, 22, 11]

    sorter = SelectionSort()

    sorted_arr = sorter.selection_sort(arr)

    for num in sorted_arr:
        print(num, end=" ")
"""